src/annotator/index.js

```diff
@@ -34,32 +34,49 @@ const appLinkEl = /** @type {Element} */ (
 );
 
 function init() {
+  window_.__hypothesis = {};
+
   const annotatorConfig = getConfig('annotator');
   const isPDF = typeof window_.PDFViewerApplication !== 'undefined';
 
-  if (annotatorConfig.subFrameIdentifier) {
-    // Other modules use this to detect if this
-    // frame context belongs to hypothesis.
-    // Needs to be a global property that's set.
-    window_.__hypothesis_frame = true;
-  }
-
   const eventBus = new EventBus();
   const guest = new Guest(document.body, eventBus, {
     ...annotatorConfig,
     // Load the PDF anchoring/metadata integration.
     // nb. documentType is an internal config property only
     documentType: isPDF ? 'pdf' : 'html',
   });
+
   const sidebar = !annotatorConfig.subFrameIdentifier
     ? new Sidebar(document.body, eventBus, guest, getConfig('sidebar'))
     : null;
+  if (sidebar) {
+    // Expose sidebar window reference for use by same-origin guest frames.
+    window_.__hypothesis.sidebarWindow = sidebar.sidebarWindow;
+  }
+
   // Clear `annotations` value from the notebook's config to prevent direct-linked
   // annotations from filtering the threads.
   const notebook = new Notebook(document.body, eventBus, getConfig('notebook'));
 
+  // Setup guest <-> sidebar communication.
+  const sidebarWindow = sidebar
+    ? sidebar.sidebarWindow
+    : /** @type {HypothesisWindow} */ (window.parent).__hypothesis
+        ?.sidebarWindow;
+  if (sidebarWindow) {
+    guest.crossframe.connectToSidebar(sidebarWindow);
```

Comment:
I appreciate that this is a separate step from `CrossFrame` object instantiation—it feels right 👍🏻